src/test/java/teammates/test/BaseTestCaseWithLocalSqlDatabaseAccess.java

```diff
@@ -0,0 +1,48 @@
+package teammates.test;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import teammates.common.datatransfer.sqlattributes.CourseAttributes;
+import teammates.common.util.HibernateUtil;
+import teammates.logic.api.LogicNewExtension;
+
+import javax.sql.DataSource;
+
+public abstract class BaseTestCaseWithLocalSqlDatabaseAccess extends BaseTestCaseWithSqlDatabaseAccess {
+    private final LogicNewExtension logic = new LogicNewExtension();
+    protected static PostgreSQLContainer pgsql;
+
+    @BeforeClass
+    public static void startContainer() {
+        pgsql = new PostgreSQLContainer<>("postgres:14.4")
+                .withDatabaseName("teammates")
+                .withUsername("teammates")
+                .withPassword("teammates");
+        pgsql.withInitScript("sql/V1__Create_course_table.sql");
```

Comment:
Eventually we should run the migration script using Flyway/Liquibase, so that there's no need to sync the migration files in `src/main/resources/db/migration` and `src/test/resources/sql`, but can leave that to another issue

Reply:
After #30 is merged can call `getFlywayInst(...).migrate()` to run the migrations.